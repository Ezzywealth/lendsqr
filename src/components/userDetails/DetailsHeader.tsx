import React from 'react';
import TableHook from '../users/Table/Hook/TableHook';

type Props = {
	id: string;
};
const DetailsHeader = ({ id }: Props) => {
	const { handleActivateUser, handleBlacklistUser } = TableHook();
	return (
		<div className='header_container'>
			<h2 className='header_title'>User Details</h2>
			<section className='btn_container'>
				<button className='btn blacklist_btn' onClick={() => handleBlacklistUser(id)}>
					Blacklist User
				</button>
				<button className='btn activate_btn' onClick={() => handleActivateUser(id)}>
					Activate User
				</button>
			</section>
		</div>
	);
};

export default DetailsHeader;
