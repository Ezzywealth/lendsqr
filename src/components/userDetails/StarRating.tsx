import React from 'react';
import { FaStar, FaRegStar } from 'react-icons/fa';

type Props = {
	rating?: number;
};

const StarRating = ({ rating=0 }: Props) => {

	const filledStars = Math.floor(rating);
	const remainder = 3 - rating;

	const renderStars = () => {
		const stars = [];

		for (let i = 0; i < filledStars; i++) {
			stars.push(<FaStar key={i} className='star filled' />);
		}

		if (remainder > 0) {
			stars.push(<FaRegStar color='#ffcf24' key='half' className='star ' />);
		}

		for (let i = filledStars + 1; i < 3; i++) {
			stars.push(<FaRegStar color='#ffcf24' key={i} className='star' />);
		}
		return stars.filter((item) => item !== undefined);
	};
	return <div className='star-rating'>{renderStars()}</div>;
};

export default StarRating;
