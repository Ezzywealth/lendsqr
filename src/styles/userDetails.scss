@import '_mixin.scss';

$primary_color: #213f7d;
$button_color: #39cdcc;
$text_color: #545f7d;
$primary_bg_color: #f5f6fa;
$login_bg_color: #fff;
$secondary_color: #545f7d;
$input_border_color: rgba(#545f7d, 0.15);
$font_family: 'Work Sans', sans-serif;
$active_bg_color: #e8fafa;
$text_black: rgba(#000, 0.8);

html,
body {
	font-family: $font_family;
}

.header_container {
	& {
		display: flex;
		justify-content: space-between;
		align-items: center;
		padding: 1rem 0;
		gap: 10px;
		box-sizing: border-box;
	}
}

.back_arrow {
	& {
		display: flex;
		gap: 10px;
		align-items: center;
		color: $text_color;
		font-size: 1rem;
		font-family: $font_family;
		cursor: pointer;
	}
}

.header_title {
	& {
		color: $primary_color;
		font-size: 1.5rem;
		font-weight: 500;
		font-family: $font_family;
	}
}

.btn_container {
	& {
		display: flex;
		gap: 10px;
		align-items: center;
	}
}

.btn {
	& {
		@include btn(transparent, transparent, 150px);
		cursor: pointer;
	}
}

.blacklist_btn {
	& {
		border: 1px solid #e4033b;
		color: #e4033b;
	}
}

.activate_btn {
	& {
		border: 1px solid $button_color;
		color: $button_color;
	}
}

.star_rating {
	& {
		display: flex;
		align-items: center;
		justify-content: flex-start;
		font-size: 16px;
	}
}

.star {
	& {
		margin-right: 4px;
		color: #ccc;
	}
}

.star.filled {
	& {
		color: #ffcf24;
	}
}

.star.half {
	& {
		color: #ffcf24;
	}
}

.user_image {
	& {
		background: #ccc;
		height: 80px;
		width: 80px;
		border-radius: 50%;
	}
}

.profile_container {
	& {
		background: $login_bg_color;
		padding: 30px 30px 0px 30px;
		width: 100%;
		box-sizing: border-box;
	}
}

.profile_top {
	& {
		display: flex;
		align-items: center;
		gap: 30px;
		box-sizing: border-box;
	}
}

.image_name {
	& {
		display: flex;
		align-items: center;
		gap: 20px;
	}
}

.name_id {
	& {
		display: flex;
		flex-direction: column;
		gap: 8px;
		h3 {
			& {
				margin: 0;
				color: $primary_color;
				font-weight: 500;
				font-size: 1.375rem;
			}
		}

		p {
			& {
				margin: 0;
				color: $text_color;
				font-size: 0.875rem;
			}
		}
	}
}

.divider {
	& {
		width: 1px;
		background-color: #ccc;
		height: 50px;
		margin: 0 10px;
	}
}

.user_tier_container {
	& {
		display: flex;
		flex-direction: column;
		gap: 10px;

		h3 {
			& {
				margin: 0;
				color: $primary_color;
				font-size: 0.875rem;
				font-weight: 500;
			}
		}
	}
}

.acc_balance_container {
	& {
		display: flex;
		flex-direction: column;
		gap: 10px;

		.acc_balance {
			& {
				color: $primary_color;
				font-weight: 500;
				font-size: 1.375rem;
				margin: 0;
				display: flex;
				align-items: center;
			}
		}

		.naira_icon {
			& {
				font-size: 1.375rem;
			}
		}

		p {
			& {
				margin: 0;
				color: $text_color;
				font-size: 0.875rem;
			}
		}

		.acc_number_name {
			& {
				display: flex;
				gap: 2px;
				color: $primary_color;
				font-size: 0.875rem;
			}
		}
	}
}

/*********************Tabs styles ************************************/

.profile_bottom {
	& {
		margin-top: 50px;
	}
}

.tabs_list {
	& {
		display: flex;
		justify-content: space-between;
		width: 100%;
		margin: 0;
		padding: 0;
		text-decoration: none;
	}

	.tab {
		& {
			text-decoration: none;
			list-style: none;
		}
	}

	.active_tab {
		& {
			color: $button_color;
			font-weight: 500;
			font-size: 1rem;
			border-bottom: 2px solid $button_color;
			padding-bottom: 10px;
			font-family: $font_family;
		}
	}

	.inactive_tab {
		& {
			color: $text_black;
			font-weight: 500;
			font-size: 1rem;
			font-family: 'SF Compact Text';
		}
	}
}

/*****************************Details Information styles***************************/
.general_informations_container {
	& {
		margin-top: 30px;
		background: $login_bg_color;
		padding: 30px;
	}

	h3,
	p {
		& {
			margin: 0;
		}
	}

	ul {
		& {
			list-style: none;
			padding: 0;
			margin: 0;
		}
	}
}

.information_section {
	& {
		display: flex;
		flex-direction: column;
		gap: 30px;
		border-bottom: 1px solid #ccc;
		padding: 30px 0;
	}
}

.guarantor {
	& {
		display: flex;
		flex-direction: column;
		border-bottom: none;
		padding: 30px 0;
	}
}

.section_details {
	& {
		display: grid;
		grid-template-columns: repeat(4, minmax(0, 1fr));
		justify-content: space-between;
		row-gap: 30px;
	}
}

.information_title {
	& {
		color: $primary_color;
		font-size: 1rem;
		font-weight: 500;
	}
}

.details_title {
	& {
		color: $text_color;
		font-size: 0.75rem;
		font-weight: 400;
		font-family: $font_family;
	}
}

.details_value {
	& {
		color: $text_color;
		font-size: 1rem;
		font-weight: 500;
		font-family: $font_family;
	}
}

.monthly_income {
	& {
		font-size: 0.5rem;
	}
}

.details_container {
	& {
		display: flex;
		flex-direction: column;
		gap: 8px;
	}
}
