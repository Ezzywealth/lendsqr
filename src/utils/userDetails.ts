import { CardProps } from '../interfaces/typings';

export const cardDetails: CardProps[] = [
	{
		id: 1,
		icon: '/assets/icons/users/users.png',
		title: 'users',
		amount: 2453,
	},
	{
		id: 2,
		icon: '/assets/icons/users/active_users.png',
		title: 'active users',
		amount: 2453,
	},
	{
		id: 3,
		icon: '/assets/icons/users/loan_users.png',
		title: 'users with loans',
		amount: 12453,
	},
	{
		id: 4,
		icon: '/assets/icons/users/savings_users.png',
		title: 'users with savings',
		amount: 102453,
	},
];

export const tableHeaders = ['organization', 'username', 'email', 'phone number', 'date joined', 'status'];

export const usersData = [
	{
		id: 1,
		organization: 'Lendsqr',
		username: 'Adedeji',
		email: 'adedeji@lendsqr.com',
		date_joined: 'May 15, 2020 10:00 AM',
		status: 'Inactive',
		phone_number: '08078903721',
	},
	{
		id: 2,
		organization: 'irorun',
		username: 'Debby Ogana',
		email: 'debby2@irorun.com',
		date_joined: 'Apr 30, 2020 10:00 AM',
		status: 'Pending',
		phone_number: '08160780928',
	},
	{
		id: 3,
		organization: 'lendstar',
		username: 'Grace Effiom',
		email: 'grace@lendstar.com',
		date_joined: 'Apr 30, 2020 10:00 AM',
		status: 'Blacklisted',
		phone_number: '07060780922',
	},
	{
		id: 4,
		organization: 'lendsqr',
		username: 'Tosin Dokunmu',
		email: 'tosin@lendsqr.com',
		date_joined: 'Apr 10, 2020 10:00 AM',
		status: 'Pending',
		phone_number: '07003309226',
	},
	{
		id: 5,
		organization: 'lendstar',
		username: 'Grace Effiom',
		email: 'grace@lendstar.com',
		date_joined: 'Apr 30, 2020 10:00 AM',
		status: 'Active',
		phone_number: '07060780922',
	},
	{
		id: 6,
		organization: 'lendsqr',
		username: 'Tosin Dokunmu',
		email: 'tosin@lendsqr.com',
		date_joined: 'Apr 10, 2020 10:00 AM',
		status: 'Active',
		phone_number: '08060780900',
	},
	{
		id: 7,
		organization: 'irorun',
		username: 'Grace Effiom',
		email: 'grace@lendstar.com',
		date_joined: 'Apr 30, 2020 10:00 AM',
		status: 'Blacklisted',
		phone_number: '07060780922',
	},
	{
		id: 8,
		organization: 'lendsqr',
		username: 'Tosin Dokunmu',
		email: 'tosin@lendsqr.com',
		date_joined: 'Apr 10, 2020 10:00 AM',
		status: 'Inactive',
		phone_number: '08060780900',
	},
	{
		id: 9,
		organization: 'lendstar',
		username: 'Grace Effiom',
		email: 'grace@lendstar.com',
		date_joined: 'Apr 30, 2020 10:00 AM',
		status: 'Inactive',
		phone_number: '07060780922',
	},
	{
		id: 10,
		organization: 'Lendsqr',
		username: 'Adedeji',
		email: 'adedeji@lendsqr.com',
		date_joined: 'May 15, 2020 10:00 AM',
		status: 'Inactive',
		phone_number: '08078903721',
	},
	{
		id: 11,
		organization: 'irorun',
		username: 'Debby Ogana',
		email: 'debby2@irorun.com',
		date_joined: 'Apr 30, 2020 10:00 AM',
		status: 'Pending',
		phone_number: '08160780928',
	},
	{
		id: 12,
		organization: 'lendstar',
		username: 'Grace Effiom',
		email: 'grace@lendstar.com',
		date_joined: 'Apr 30, 2020 10:00 AM',
		status: 'Blacklisted',
		phone_number: '07060780922',
	},
	{
		id: 13,
		organization: 'lendsqr',
		username: 'Tosin Dokunmu',
		email: 'tosin@lendsqr.com',
		date_joined: 'Apr 10, 2020 10:00 AM',
		status: 'Pending',
		phone_number: '07003309226',
	},
	{
		id: 14,
		organization: 'lendstar',
		username: 'Grace Effiom',
		email: 'grace@lendstar.com',
		date_joined: 'Apr 30, 2020 10:00 AM',
		status: 'Active',
		phone_number: '07060780922',
	},
	{
		id: 15,
		organization: 'lendsqr',
		username: 'Tosin Dokunmu',
		email: 'tosin@lendsqr.com',
		date_joined: 'Apr 10, 2020 10:00 AM',
		status: 'Active',
		phone_number: '08060780900',
	},
	{
		id: 16,
		organization: 'irorun',
		username: 'Grace Effiom',
		email: 'grace@lendstar.com',
		date_joined: 'Apr 30, 2020 10:00 AM',
		status: 'Blacklisted',
		phone_number: '07060780922',
	},
	{
		id: 17,
		organization: 'lendsqr',
		username: 'Tosin Dokunmu',
		email: 'tosin@lendsqr.com',
		date_joined: 'Apr 10, 2020 10:00 AM',
		status: 'Inactive',
		phone_number: '08060780900',
	},
	{
		id: 18,
		organization: 'lendstar',
		username: 'Grace Effiom',
		email: 'grace@lendstar.com',
		date_joined: 'Apr 30, 2020 10:00 AM',
		status: 'Inactive',
		phone_number: '07060780922',
	},
	{
		id: 19,
		organization: 'Lendsqr',
		username: 'Adedeji',
		email: 'adedeji@lendsqr.com',
		date_joined: 'May 15, 2020 10:00 AM',
		status: 'Inactive',
		phone_number: '08078903721',
	},
	{
		id: 20,
		organization: 'irorun',
		username: 'Debby Ogana',
		email: 'debby2@irorun.com',
		date_joined: 'Apr 30, 2020 10:00 AM',
		status: 'Pending',
		phone_number: '08160780928',
	},
	{
		id: 21,
		organization: 'lendstar',
		username: 'Grace Effiom',
		email: 'grace@lendstar.com',
		date_joined: 'Apr 30, 2020 10:00 AM',
		status: 'Blacklisted',
		phone_number: '07060780922',
	},
	{
		id: 22,
		organization: 'lendsqr',
		username: 'Tosin Dokunmu',
		email: 'tosin@lendsqr.com',
		date_joined: 'Apr 10, 2020 10:00 AM',
		status: 'Pending',
		phone_number: '07003309226',
	},
	{
		id: 23,
		organization: 'lendstar',
		username: 'Grace Effiom',
		email: 'grace@lendstar.com',
		date_joined: 'Apr 30, 2020 10:00 AM',
		status: 'Active',
		phone_number: '07060780922',
	},
	{
		id: 24,
		organization: 'lendsqr',
		username: 'Tosin Dokunmu',
		email: 'tosin@lendsqr.com',
		date_joined: 'Apr 10, 2020 10:00 AM',
		status: 'Active',
		phone_number: '08060780900',
	},
	{
		id: 25,
		organization: 'irorun',
		username: 'Grace Effiom',
		email: 'grace@lendstar.com',
		date_joined: 'Apr 30, 2020 10:00 AM',
		status: 'Blacklisted',
		phone_number: '07060780922',
	},
	{
		id: 26,
		organization: 'lendsqr',
		username: 'Tosin Dokunmu',
		email: 'tosin@lendsqr.com',
		date_joined: 'Apr 10, 2020 10:00 AM',
		status: 'Inactive',
		phone_number: '08060780900',
	},
	{
		id: 27,
		organization: 'lendstar',
		username: 'Grace Effiom',
		email: 'grace@lendstar.com',
		date_joined: 'Apr 30, 2020 10:00 AM',
		status: 'Inactive',
		phone_number: '07060780922',
	},
	{
		id: 28,
		organization: 'Lendsqr',
		username: 'Adedeji',
		email: 'adedeji@lendsqr.com',
		date_joined: 'May 15, 2020 10:00 AM',
		status: 'Inactive',
		phone_number: '08078903721',
	},
	{
		id: 29,
		organization: 'irorun',
		username: 'Debby Ogana',
		email: 'debby2@irorun.com',
		date_joined: 'Apr 30, 2020 10:00 AM',
		status: 'Pending',
		phone_number: '08160780928',
	},
	{
		id: 30,
		organization: 'lendstar',
		username: 'Grace Effiom',
		email: 'grace@lendstar.com',
		date_joined: 'Apr 30, 2020 10:00 AM',
		status: 'Blacklisted',
		phone_number: '07060780922',
	},
	{
		id: 31,
		organization: 'lendsqr',
		username: 'Tosin Dokunmu',
		email: 'tosin@lendsqr.com',
		date_joined: 'Apr 10, 2020 10:00 AM',
		status: 'Pending',
		phone_number: '07003309226',
	},
	{
		id: 32,
		organization: 'lendstar',
		username: 'Grace Effiom',
		email: 'grace@lendstar.com',
		date_joined: 'Apr 30, 2020 10:00 AM',
		status: 'Active',
		phone_number: '07060780922',
	},
	{
		id: 33,
		organization: 'lendsqr',
		username: 'Tosin Dokunmu',
		email: 'tosin@lendsqr.com',
		date_joined: 'Apr 10, 2020 10:00 AM',
		status: 'Active',
		phone_number: '08060780900',
	},
	{
		id: 34,
		organization: 'irorun',
		username: 'Grace Effiom',
		email: 'grace@lendstar.com',
		date_joined: 'Apr 30, 2020 10:00 AM',
		status: 'Blacklisted',
		phone_number: '07060780922',
	},
	{
		id: 35,
		organization: 'lendsqr',
		username: 'Tosin Dokunmu',
		email: 'tosin@lendsqr.com',
		date_joined: 'Apr 10, 2020 10:00 AM',
		status: 'Inactive',
		phone_number: '08060780900',
	},
	{
		id: 36,
		organization: 'lendstar',
		username: 'Grace Effiom',
		email: 'grace@lendstar.com',
		date_joined: 'Apr 30, 2020 10:00 AM',
		status: 'Inactive',
		phone_number: '07060780922',
	},
	{
		id: 37,
		organization: 'Lendsqr',
		username: 'Adedeji',
		email: 'adedeji@lendsqr.com',
		date_joined: 'May 15, 2020 10:00 AM',
		status: 'Inactive',
		phone_number: '08078903721',
	},
	{
		id: 38,
		organization: 'irorun',
		username: 'Debby Ogana',
		email: 'debby2@irorun.com',
		date_joined: 'Apr 30, 2020 10:00 AM',
		status: 'Pending',
		phone_number: '08160780928',
	},
	{
		id: 39,
		organization: 'lendstar',
		username: 'Grace Effiom',
		email: 'grace@lendstar.com',
		date_joined: 'Apr 30, 2020 10:00 AM',
		status: 'Blacklisted',
		phone_number: '07060780922',
	},
	{
		id: 40,
		organization: 'lendsqr',
		username: 'Tosin Dokunmu',
		email: 'tosin@lendsqr.com',
		date_joined: 'Apr 10, 2020 10:00 AM',
		status: 'Pending',
		phone_number: '07003309226',
	},
	{
		id: 41,
		organization: 'lendstar',
		username: 'Grace Effiom',
		email: 'grace@lendstar.com',
		date_joined: 'Apr 30, 2020 10:00 AM',
		status: 'Active',
		phone_number: '07060780922',
	},
	{
		id: 42,
		organization: 'lendsqr',
		username: 'Tosin Dokunmu',
		email: 'tosin@lendsqr.com',
		date_joined: 'Apr 10, 2020 10:00 AM',
		status: 'Active',
		phone_number: '08060780900',
	},
	{
		id: 43,
		organization: 'irorun',
		username: 'Grace Effiom',
		email: 'grace@lendstar.com',
		date_joined: 'Apr 30, 2020 10:00 AM',
		status: 'Blacklisted',
		phone_number: '07060780922',
	},
	{
		id: 44,
		organization: 'lendsqr',
		username: 'Tosin Dokunmu',
		email: 'tosin@lendsqr.com',
		date_joined: 'Apr 10, 2020 10:00 AM',
		status: 'Inactive',
		phone_number: '08060780900',
	},
	{
		id: 46,
		organization: 'lendstar',
		username: 'Grace Effiom',
		email: 'grace@lendstar.com',
		date_joined: 'Apr 30, 2020 10:00 AM',
		status: 'Inactive',
		phone_number: '07060780922',
	},
];
