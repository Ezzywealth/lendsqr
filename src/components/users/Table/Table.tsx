import React from 'react';
import { tableHeaders, usersData } from '../../../utils/userDetails';
import { BsFilter, BsThreeDotsVertical } from 'react-icons/bs';

const Table = () => {
	return (
		<div className='table-responsive'>
			<table className='custom-table'>
				<thead>
					<tr>
						{tableHeaders.map((header, index) => {
							return (
								<th key={index}>
									{header}
									<img src='/assets/icons/users/filter-button.png' alt='filter' className='filter-icon' />
								</th>
							);
						})}
					</tr>
				</thead>
				<tbody>
					{usersData.slice(0, 9).map((user, index) => {
						return (
							<tr key={index}>
								<td>{user.organization}</td>
								<td>{user.username}</td>
								<td>{user.email}</td>
								<td>{user.phone_number}</td>
								<td>{user.date_joined}</td>
								<td>{user.status}</td>
								<td>
									<BsThreeDotsVertical size={20} />
								</td>
							</tr>
						);
					})}
				</tbody>
			</table>
		</div>
	);
};

export default Table;
