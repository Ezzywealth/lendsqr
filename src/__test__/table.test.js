import React from 'react';
import { render, screen, fireEvent, act } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store'; // You may need to install this package
import { usersData } from '../utils/userDetails';
import Table from '../components/users/Table/Table';

const mockStore = configureStore();
jest.mock('../utils/userDetails.ts', () => ({
	tableHeaders: ['organization', 'username', 'email', 'phone number', 'date joined', 'status'],
	filteredUsers: [
		{
			'customId': '6550b257581b3a7f0',
			'id': 1,
			'profile_picture': 'https://picsum.photos/200/300?random=446',
			'organization': 'Lendstar',
			'username': 'Snider92',
			'fullName': 'Mcclain Head',
			'email': 'mcclainhead@bitendrex.com',
			'date_joined': 'Aug 09, 2020 03:57 AM',
			'status': 'Active',
			'phone_number': '+1 (989) 466-2279',
			'acc_tier': 3,
			'acc_balance': 13342819,
			'acc_number': 21704038798,
			'bank_name': 'VURBO',
			'bvn': 2222663344,
			'gender': 'Female',
			'marital_status': 'Divorced',
			'children': 'Yes',
			'resident_type': 'Parental Resident',
			'education_level': "Master's Degree",
			'employment_status': 'Employed',
			'industry': 'Technology',
			'duration_of_employment': '2 years',
			'office_email': 'mcclainhead@vurbo.com',
			'monthly_income': {
				'min': 3103,
				'max': 10000,
			},
			'loan_repayment': 491.40000000000003,
			'socials': [
				{
					'platform': 'facebook',
					'handle': 'Graciela Cardenas',
				},
				{
					'platform': 'twitter',
					'handle': '@Maryann30',
				},
				{
					'platform': 'instagram',
					'handle': '@Agnes59',
				},
			],
			'guarantor': [
				{
					'fullName': 'Gentry Clay',
					'phone_number': '+1 (927) 417-2383',
					'email': 'gentryclay@vurbo.com',
					'relationship': 'Colleague',
				},
				{
					'fullName': 'Massey Casey',
					'phone_number': '+1 (918) 460-3318',
					'email': 'masseycasey@vurbo.com',
					'relationship': 'Colleague',
				},
				{
					'fullName': 'Soto Burgess',
					'phone_number': '+1 (835) 427-2949',
					'email': 'sotoburgess@vurbo.com',
					'relationship': 'Friend',
				},
			],
		},
		{
			'customId': '6550b2578a3fedf31',
			'id': 2,
			'profile_picture': 'https://picsum.photos/200/300?random=559',
			'organization': 'Lendsqr',
			'username': 'Carey77',
			'fullName': 'Christina Rush',
			'email': 'christinarush@vurbo.com',
			'date_joined': 'Jan 16, 2020 11:15 PM',
			'status': 'Pending',
			'phone_number': '+1 (897) 587-3994',
			'acc_tier': 3,
			'acc_balance': 72290141,
			'acc_number': 37068730901,
			'bank_name': 'MAXEMIA',
			'bvn': 2222690725,
			'gender': 'Female',
			'marital_status': 'Single',
			'children': 'Yes',
			'resident_type': 'Own Resident',
			'education_level': "Bachelor's Degree",
			'employment_status': 'Employed',
			'industry': 'Finance',
			'duration_of_employment': '2 years',
			'office_email': 'christinarush@maxemia.com',
			'monthly_income': {
				'min': 5052,
				'max': 10000,
			},
			'loan_repayment': 600.1,
			'socials': [
				{
					'platform': 'facebook',
					'handle': 'Benson Cochran',
				},
				{
					'platform': 'twitter',
					'handle': '@Howe76',
				},
				{
					'platform': 'instagram',
					'handle': '@Schmidt76',
				},
			],
			'guarantor': [
				{
					'fullName': 'Reeves Woods',
					'phone_number': '+1 (955) 530-2008',
					'email': 'reeveswoods@maxemia.com',
					'relationship': 'Family',
				},
				{
					'fullName': 'Gallegos Hurley',
					'phone_number': '+1 (882) 590-3045',
					'email': 'gallegoshurley@maxemia.com',
					'relationship': 'Friend',
				},
				{
					'fullName': 'Benton Dean',
					'phone_number': '+1 (884) 477-3410',
					'email': 'bentondean@maxemia.com',
					'relationship': 'Family',
				},
			],
		},
		{
			'customId': '6550b25742f7956c2',
			'id': 3,
			'profile_picture': 'https://picsum.photos/200/300?random=214',
			'organization': 'Lendstar',
			'username': 'Tessa31',
			'fullName': 'Kimberly Wall',
			'email': 'kimberlywall@maxemia.com',
			'date_joined': 'Dec 13, 2022 04:31 AM',
			'status': 'Blacklisted',
			'phone_number': '+1 (886) 534-2090',
			'acc_tier': 3,
			'acc_balance': 99585584,
			'acc_number': 57031766312,
			'bank_name': 'QUALITEX',
			'bvn': 2222679749,
			'gender': 'Male',
			'marital_status': 'Married',
			'children': 'No',
			'resident_type': 'Own Resident',
			'education_level': "Bachelor's Degree",
			'employment_status': 'Unemployed',
			'industry': 'Technology',
			'duration_of_employment': '1 year',
			'office_email': 'kimberlywall@qualitex.com',
			'monthly_income': {
				'min': 9712,
				'max': 10000,
			},
			'loan_repayment': 952.5,
			'socials': [
				{
					'platform': 'facebook',
					'handle': 'Hooper Frederick',
				},
				{
					'platform': 'twitter',
					'handle': '@Luann98',
				},
				{
					'platform': 'instagram',
					'handle': '@Kennedy68',
				},
			],
			'guarantor': [
				{
					'fullName': 'Hess Black',
					'phone_number': '+1 (909) 473-2937',
					'email': 'hessblack@qualitex.com',
					'relationship': 'Family',
				},
				{
					'fullName': 'Leon Madden',
					'phone_number': '+1 (828) 573-2217',
					'email': 'leonmadden@qualitex.com',
					'relationship': 'Friend',
				},
				{
					'fullName': 'Annette Ayers',
					'phone_number': '+1 (960) 502-2086',
					'email': 'annetteayers@qualitex.com',
					'relationship': 'Family',
				},
			],
		},
	],
}));
describe('Table component', () => {
	let store;

	beforeEach(() => {
		// Initial state for your Redux store
		const initialState = {
			users: {
				filteredUsers: [],
				currentPage: 1,
				totalPages: 3,
				pageSize: 10,
				noOfItems: 25,
				showOptionsModal: false,
				showFilterModal: false,
			},
		};

		store = mockStore(initialState);
	});

	test('renders LoginForm component', () => {
		const { container } = render(
			<Provider store={store}>
				<Table />
			</Provider>
		);
		expect(container).toMatchSnapshot();
	});

	it('renders table headers', () => {
		render(
			<Provider store={store}>
				<Table />
			</Provider>
		);

		// Add assertions for each header
		expect(screen.getByText('Organization')).toBeInTheDocument();
		expect(screen.getByText('Username')).toBeInTheDocument();
		// Add assertions for other headers
	});

	it('renders user data', () => {
		render(
			<Provider store={store}>
				<Table />
			</Provider>
		);

		// Add assertions for user data rendering
		// For example, check if a user's organization is displayed
		expect(screen.getByText('Organization')).toBeInTheDocument();
		// Add assertions for other user data
	});

	it('opens filter modal when filter icon is clicked', async () => {
		render(
			<Provider store={store}>
				<Table />
			</Provider>
		);

		// Click on the filter icon
		fireEvent.click(screen.getByTestId('filter-1'));

		// Wait for the modal to be visible (use async/await if needed)
		await screen.findByText('Organization');

		// Add assertions for the visible modal
		expect(screen.getByText('Organization')).toBeInTheDocument();
	});
});
