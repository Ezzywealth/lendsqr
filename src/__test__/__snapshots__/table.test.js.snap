// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component renders LoginForm component 1`] = `
<div>
  <div>
    <div
      class="table-responsive"
    >
      <table
        class="custom-table"
      >
        <thead>
          <tr>
            <th>
              <div
                class="filter_container"
              >
                organization
                <img
                  alt="filter"
                  class="filter-icon"
                  data-testid="filter-0"
                  src="/assets/icons/users/filter-button.png"
                />
              </div>
            </th>
            <th>
              <div
                class="filter_container"
              >
                username
                <img
                  alt="filter"
                  class="filter-icon"
                  data-testid="filter-1"
                  src="/assets/icons/users/filter-button.png"
                />
              </div>
            </th>
            <th>
              <div
                class="filter_container"
              >
                email
                <img
                  alt="filter"
                  class="filter-icon"
                  data-testid="filter-2"
                  src="/assets/icons/users/filter-button.png"
                />
              </div>
            </th>
            <th>
              <div
                class="filter_container"
              >
                phone number
                <img
                  alt="filter"
                  class="filter-icon"
                  data-testid="filter-3"
                  src="/assets/icons/users/filter-button.png"
                />
              </div>
            </th>
            <th>
              <div
                class="filter_container"
              >
                date joined
                <img
                  alt="filter"
                  class="filter-icon"
                  data-testid="filter-4"
                  src="/assets/icons/users/filter-button.png"
                />
              </div>
            </th>
            <th>
              <div
                class="filter_container"
              >
                status
                <img
                  alt="filter"
                  class="filter-icon"
                  data-testid="filter-5"
                  src="/assets/icons/users/filter-button.png"
                />
              </div>
            </th>
          </tr>
        </thead>
        <tbody
          class="table-body"
        >
          <tr
            class="hide-modal"
          >
            <td
              colspan="3"
            >
              <div
                class="modal show"
              >
                <div
                  class="modal-content"
                >
                  <label>
                    Organization
                    <select>
                      <option
                        disabled=""
                        value=""
                      >
                        Select
                      </option>
                      <option
                        class="options"
                        value="lendsqr"
                      >
                        lendsqr
                      </option>
                      <option
                        class="options"
                        value="lendstar"
                      >
                        lendstar
                      </option>
                      <option
                        class="options"
                        value="irorun"
                      >
                        irorun
                      </option>
                    </select>
                  </label>
                  <label>
                    Username
                    <input
                      placeholder="User"
                      type="text"
                      value=""
                    />
                  </label>
                  <label>
                    Email
                    <input
                      pattern="[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\\\.[a-zA-Z]{2,}"
                      placeholder="Email"
                      required=""
                      type="email"
                      value=""
                    />
                  </label>
                  <label>
                    Date
                    <input
                      placeholder="Date"
                      type="date"
                      value=""
                    />
                  </label>
                  <label>
                    Phone Number
                    <input
                      placeholder="Phone"
                      type="text"
                      value=""
                    />
                  </label>
                  <label>
                    Status
                    <select>
                      <option
                        class="options"
                        disabled=""
                        value=""
                      >
                        Select
                      </option>
                      <option
                        class="options"
                        value="active"
                      >
                        Active
                      </option>
                      <option
                        class="options"
                        value="pending"
                      >
                        Pending
                      </option>
                      <option
                        class="options"
                        value="inactive"
                      >
                        Inactive
                      </option>
                      <option
                        class="options"
                        value="blacklisted"
                      >
                        Blacklisted
                      </option>
                    </select>
                  </label>
                  <div
                    class="button-container"
                  >
                    <button
                      class="btn cancel_button"
                    >
                      Cancel
                    </button>
                    <button
                      class="btn apply_button"
                    >
                      Apply
                    </button>
                  </div>
                </div>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="pagination-container"
    >
      <div
        class="pagination-info"
      >
        <p>
          Showing
        </p>
        <select
          class="select-field"
        >
          <option
            data-testid="option-1"
            value="1"
          >
            1
          </option>
          <option
            data-testid="option-2"
            value="2"
          >
            2
          </option>
          <option
            data-testid="option-3"
            value="3"
          >
            3
          </option>
        </select>
        out of 
        3
      </div>
      <div
        class="pagination-pages"
      >
        <span
          class="pagination-arrow"
        >
          <svg
            fill="currentColor"
            height="25"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
            />
          </svg>
        </span>
        <button
          class="pagination-page active"
        >
          1
        </button>
        <button
          class="pagination-page "
        >
          2
        </button>
        <button
          class="pagination-page "
        >
          3
        </button>
        <button
          class="pagination-page "
        >
          2
        </button>
        <button
          class="pagination-page "
        >
          3
        </button>
        <span
          class="pagination-arrow"
        >
          <svg
            fill="currentColor"
            height="25"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</div>
`;
